test/Microsoft.IdentityModel.TestUtils/TokenValidationExtensibility/Tests/ExtensibilityTheoryData.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Security.Claims;
+using System;
+using Microsoft.IdentityModel.Tokens;
+
+#nullable enable
+namespace Microsoft.IdentityModel.TestUtils.TokenValidationExtensibility.Tests
+{
+    public class ExtensibilityTheoryData : TheoryDataBase
+    {
+        string _tokenHandlerType;
+        SecurityTokenDescriptor? _securityTokenDescriptor;
+
+        internal ExtensibilityTheoryData(
+            string testId,
+            string tokenHandlerType,
+            int extraStackFrames) : base(testId)
+        {
+            ExtraStackFrames = extraStackFrames;
+            TokenHandler = SecurityTokenHandlerForType(tokenHandlerType);
+            _tokenHandlerType = tokenHandlerType;
+
+            ValidationParameters = ValidationParametersSkippingValidations();
+        }
+
+        private IExtensibilityTestingTokenHandler SecurityTokenHandlerForType(string tokenHandlerType)
+        {
+            return tokenHandlerType switch
+            {
+                "JWT" => new JsonWebTokenHandlerWithResult(),
+                "SAML" => new SamlSecurityTokenHandlerWithResult(),
+                "SAML2" => new Saml2SecurityTokenHandlerWithResult(),
+                _ => throw new NotImplementedException(tokenHandlerType)
+            };
+        }
+
+        private SecurityTokenDescriptor PopulateSubjectForSecurityTokenDescriptor(
```

Comment:
```suggestion
        private SecurityTokenDescriptor CreateSecurityTokenDescriptor(
```

Reply:
The other `Create` I think are on point, but in this one we are just adding the `Subject` to an existing `SecurityTokenDescriptor` simply because it is different between JWT and SAML.
We still want the caller to configure the rest of the `SecurityTokenDescriptor` before reaching this point.